### ISSUE TITLE
[💡SUG] Support for dual-target or multi-target cross-domain recommendation

### ISSUE THREAD

**Issue by ShadowTinker** [RUCAIBox/RecBole-CDR#42]
**Is your feature request related to a problem? Please describe.**
Current RecBole-CDR only supports single-target CDR models. Will you add support for multi-target CDR problem mentioned in this [survey](https://arxiv.org/abs/2103.01696) in a future release? Multi-target CDR is to improve the recommendation accuracy in all domains simultaneously rather than just one target domain.

**Describe the solution you'd like**
New dataset, dataloader and trainer supporting multi-target CDR. 

**Describe alternatives you've considered**
I find this part in dataset.py. But I still don't know how to set `source_split_flag` and change single-target CDR to multi-target CDR
https://github.com/RUCAIBox/RecBole-CDR/blob/1758db49e193be4ba70b4c0023bba78af6c2a1f9/recbole_cdr/data/dataset.py#L558-L567

**Comment by Wicknight** [comment]
@ShadowTinker Hello, thanks for your attention to RecBole-CDR!
Thank you for your advice! We will consider providing support for multi-target CDR in the new release. As for the `source_split_flag` in dataset.py, we can not implement multi-target CDR with this parameter. It only provides the training data of the source domain for training, and the ultimate goal is to improve the performance of the target domain. Maybe you can try changing the source domain dataset structure while getting training, validation, test sets, and changing the evaluation method in the trainer.

**Comment by ShadowTinker** [comment]
@Wicknight Thanks for your reply and kind advice! I'm looking forward to the new release of RecBole-CDR. I'll try to change the code according to your suggestion.
